Hog Descriptor Generation

RossBrunton opened the pull request:
This is good enough for us all to review. I guess once we have handed in our report (aiming to do so on Thursday), we should all sit down and read through all of this, make comments, and then presumably fight over minor things.

And I guess since this is a big thing, we should get two "ship its" before it is merged. Also, this includes changes from #11 , so that will need to be merged before this.

... And why can't I add myself as a reviewer? How bothersome.

RossBrunton committed:
(ImageProcessing) Started work on gradient actor

It now reads in 8x8 cells properly, and can calculate their magnitude.
All that's left is to somehow calculate the angle, and write the docs!

RossBrunton committed:
(ImageProcessing) Started work on gradient actor

It now reads in 8x8 cells properly, and can calculate their magnitude.
All that's left is to somehow calculate the angle, and write the docs!

RossBrunton committed:
(ImageProcessing) The angle is now output

Or at least, it should be. There is certainly some form of maths going
on.

RossBrunton committed:
(ImageProcessing) The angle is now output

Or at least, it should be. There is certainly some form of maths going
on.

RossBrunton committed:
(ImageProcessing) Added docs for Gradient

RossBrunton committed:
(ImageProcessing) Added docs for Gradient

cajturner committed:
WIP Hog hist binning

cajturner committed:
WIP Hog hist binning

cajturner committed:
Simpler binning

cajturner committed:
Simpler binning

Teymoor-Ali committed:
added algorithm

Teymoor-Ali committed:
added algorithm

Teymoor-Ali committed:
Added Hog files

Teymoor-Ali committed:
Added Hog files

Teymoor-Ali committed:
fixed sliding window

Teymoor-Ali committed:
fixed sliding window

Teymoor-Ali committed:
Delete slidingwindowadvanced.cal

Teymoor-Ali committed:
Delete slidingwindowadvanced.cal

Teymoor-Ali committed:
Delete bigslidingwindow.cal

Teymoor-Ali committed:
Delete bigslidingwindow.cal

RossBrunton committed:
(ImageProcessing) Added Blocks actor

This gets four histogram cells and arranges them into a 2x2 squares.

RossBrunton committed:
(ImageProcessing) Added Blocks actor

This gets four histogram cells and arranges them into a 2x2 squares.

RossBrunton committed:
(ImageProcessing) Tests

RossBrunton committed:
(ImageProcessing) Tests

Teymoor-Ali committed:
Added inversesliding window

Teymoor-Ali committed:
Added inversesliding window

Teymoor-Ali committed:
Merge branch 'master' of https://github.com/Teymoor-Ali/orc-apps

RossBrunton committed:
Merge branch 'teymoor' into hogall

RossBrunton committed:
(ImageProcessing) Test script for Gradient

Also fixed a lot of bugs in it, because I'm good programmer. Mostly
silly things.

I did change the interface to it, to use two outputs instead of writing
pairs to one. Much easier to connect it to stuff then, and it makes more
sense.

RossBrunton committed:
(ImageProcessing) Test script for Gradient

Also fixed a lot of bugs in it, because I'm good programmer. Mostly
silly things.

I did change the interface to it, to use two outputs instead of writing
pairs to one. Much easier to connect it to stuff then, and it makes more
sense.

RossBrunton committed:
(ImageProcessing) Added gradient angle test

RossBrunton committed:
(ImageProcessing) Added gradient angle test

cajturner committed:
Fix Histogram binning bugs. Adds test actors

cajturner committed:
Fix Histogram binning bugs. Adds test actors

cajturner committed:
Fixes some histogram logic bugs. Adds a second test case

cajturner committed:
Fixes some histogram logic bugs. Adds a second test case

RossBrunton committed:
Merge remote-tracking branch 'upstream/Hog_binning' into hogall

cajturner committed:
Add Histogram binning doc

cajturner committed:
Add Histogram binning doc

Teymoor-Ali committed:
Added Histogram Normalisation Algorithm

Teymoor-Ali committed:
Added Histogram Normalisation Algorithm

cajturner committed:
Adding missed input file

cajturner committed:
Adding missed input file

RossBrunton committed:
Merge branch 'master' into hogall

RossBrunton committed:
Merge branch 'teymoor' into hogall

RossBrunton committed:
Merge branch 'Hog_binning' into hogall

RossBrunton committed:
(ImageProcessing) Joined all the hog stuff

Still doesn't work because the normalization stuff doesn't work right.

RossBrunton committed:
(ImageProcessing) Joined all the hog stuff

Still doesn't work because the normalization stuff doesn't work right.

Teymoor-Ali committed:
Updated Histo normalisation

Teymoor-Ali committed:
Updated Histo normalisation

RossBrunton committed:
Merge branch 'teymoor' into hogall

Teymoor-Ali committed:
Update to hist normalisation

Teymoor-Ali committed:
Update to hist normalisation

RossBrunton committed:
Merge branch 'teymoor' into hogall

Teymoor-Ali committed:
Added fix to histogram normalisation

Teymoor-Ali committed:
Added fix to histogram normalisation

RossBrunton committed:
(Image Processing) Put everything in same package

RossBrunton committed:
(Image Processing) Put everything in same package

RossBrunton committed:
Moved some constants into a constants package

RossBrunton committed:
Moved some constants into a constants package

RossBrunton committed:
(Image Processing) Consistent whitespace

Removed trailing spaces, and converted all spaces to tabs (usually from
my code, but sometimes spaces have snuck in for some reason).

RossBrunton committed:
(Image Processing) Consistent whitespace

Removed trailing spaces, and converted all spaces to tabs (usually from
my code, but sometimes spaces have snuck in for some reason).

RossBrunton committed:
Merge branch 'hogall' of github.com:robstewart57/orc-apps into hogall

Teymoor-Ali committed:
Renamed Files and fixed bug in inversetiler

Teymoor-Ali committed:
Merge branch 'hogall' of https://github.com/robstewart57/orc-apps into hogall

RossBrunton committed:
Merge branch 'master' into hogall

RossBrunton commented on ImageProcessing/src/image/hog_person_detector/hist_binning_input.cal:
```diff
@@ -0,0 +1,9 @@
+package image.hog_person_detector;
```
Should this file even be here?

RossBrunton commented on ImageProcessing/src/image/hog_person_detector/hist_binning_input2.cal:
```diff
@@ -0,0 +1,15 @@
+package image.hog_person_detector;
```
If these are all for testing, shouldn't they be in `hog_person_detector.test`?

RossBrunton commented on ImageProcessing/src/image/hog_person_detector/histogram_binning.cal:
```diff
@@ -0,0 +1,81 @@
+		read_count < 64
+	do
+		i := intense;
+		if angle <= 10 then
```
Shouldn't values less than 10 cause part of it to go into the largest bin?

RossBrunton commented on ImageProcessing/src/image/hog_person_detector/histogram_binning.cal:
```diff
@@ -0,0 +1,81 @@
+			histogram[0] := histogram[0] + i;
+		else
+			if angle >= 170 then
+				histogram[8] := histogram[8] + i;
```
And ditto with the smallest bin?

RossBrunton commented on ImageProcessing/src/image/hog_person_detector/histogram_normalisation.cal:
```diff
@@ -0,0 +1,103 @@
+	do
+		histogram_count := 0;
+	end //normalising all values in the block between 0 and 1 depending on the minimum and maximum values within the whole block
+	write: action ==> Gout:[ out ]
```
Probably add an empty line before this action for clarity

RossBrunton commented on ImageProcessing/src/image/hog_person_detector/histogram_normalisation.cal:
```diff
@@ -0,0 +1,103 @@
+		histogram_count := histogram_count + 1;
+	end
+
+	reset: action ==>
```
Doesn't reset max and min.

Teymoor-Ali replied:
Yep saw that issue, ill push an update through

RossBrunton commented on ImageProcessing/src/image/hog_person_detector/inverseslidingwindow.cal:
```diff
@@ -0,0 +1,133 @@
+
+
+		out := buffer[(y + yBase) * imageWidth + x + xBase];
+		// for x to windowDimension
```
Lots of commented out "dead" code

Teymoor-Ali replied:
dont need inversesliding window or sliding window,  removing those files

Teymoor-Ali replied:
tiler/inverse tiler are the same thing

RossBrunton commented on ImageProcessing/src/image/hog_person_detector/inverseslidingwindow.cal:
```diff
@@ -0,0 +1,133 @@
+				// then yBase++
+				// xbase == 0
+		x := x + 1;
+		if x >= windowDimension then
```
Indenting is a bit wacky here

RossBrunton commented on ImageProcessing/src/image/hog_person_detector/inversetiler.cal:
```diff
@@ -0,0 +1,121 @@
+actor inversetiler () uint(size=8) Gin ==> uint(size=8) Gout :
+
+/* Constants */
+	uint(size=8) imageHeight = 128;
```
Forgot to update this to use the constants, whoops!

RossBrunton commented on ImageProcessing/src/image/hog_person_detector/tiler.cal:
```diff
@@ -0,0 +1,118 @@
+
```
I think I've seen this before

RossBrunton reviewed (commented):
Had a look through everything, and left some comments. Some of them aren't about whitespace this time! :D

(Obligatory I suck at tone, so don't take my abusive comments to heart or whatever)

Teymoor-Ali committed:
reset the min/max and deleted not-related files

RossBrunton commented on ImageProcessing/src/image/hog_person_detector/histogram_normalisation.cal:
```diff
@@ -35,12 +35,10 @@
 
 /* Constants */
-	uint(size=8) histogram_bins := HIST_LENGTH * 4;
+	uint(size=8) histogram_bins := 36;
```
Why has this been reverted?

Teymoor-Ali replied:
oh must of copy and pasted the old one, ill fix it.

Teymoor-Ali committed:
Added previouslt deleted code

RossBrunton committed:
(ImageProcessing) Made changes to hist. binning

This basically provides a number of bugfixes and changes to the
algorithm. Mostly making it so that angles less than 10 get put into the
biggest bucket, and ones greater than 170 go into the smallest. I also
moved the testing files into the testing package.

RossBrunton commented:
Calum isn't able to work on it until tomorrow, so asked me to do his changes. Unfortunately I changed a lot of things which appeared to be bugs to me, but may not be. Calum, when you get back could you give the changes a look over and run the tests to make sure everything works?

Teymoor-Ali reviewed (commented):
Looks good,

cajturner reviewed (approved):
Changes look fine. I wasn't aware that the bins wrapped around. Test return expected output

robstewart57 commented:
@cajturner @RossBrunton so... can I squash and merge this PR now?

RossBrunton commented:
I think so, all my review comments seem to be addressed.
